app/controllers/api/v1/lookup_keys_controller.rb

```diff
@@ -57,7 +57,11 @@ def update
       param :id, :identifier, :required => true
 
       def destroy
-        process_response @lookup_key.destroy
+        if @lookup_key.type == "PuppetclassLookupKey"
```

Comment:
This doesn't give any hint to the user about what exactly is wrong. I'd say just put this and let `process_response @lookup_key.destroy` handle the 422. You can also do it in a `before_action`.

```ruby

        if @lookup_key.type == "PuppetclassLookupKey"
          @lookup_key.errors.add(:base, 'Smart class parameters cannot be destroyed')
        end
```